src/engine/SCons/Platform/__init__.py

```diff
@@ -216,8 +226,9 @@ def __call__(self, target, source, env, for_signature):
             # Windows path names.
             rm = 'del'
 
-        prefix = env.subst('$TEMPFILEPREFIX')
-        if not prefix:
+        if env.has_key('TEMPFILEPREFIX'):
+            prefix = env.subst('$TEMPFILEPREFIX')
+        else:
```

Comment:
This is not necessarily equivalent as if TEMPFILEPREFIX isn't defined it would be an empty string and trigger if not prefix..